MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js

```diff
@@ -295,7 +313,8 @@ define([
          * @param {String} nodePath
          */
         setActiveNodeFilter: function (nodePath) {
-            if (this.activeNode() === nodePath) {
+
+            if (this.activeNode() === nodePath && !this.jsTreeReloaded()) {
```

Comment:
I would prefer to avoid this approach of using property that can be changed anywhere and change the behaviour of the function, it would be more straight forward to add a parameter to this function that specifically changes its behaviour (would be better to avoid this as well).

From what I could see locally, the problem of opening the gallery multiple times and the folder getting unselected is related to:

```
        /**
         * Locate and higlight node in jstree by path id.
         *
         * @param {String} path
         */
        locateNode: function (path) {
            var selectedId =  $(this.directoryTreeSelector).jstree('get_selected').attr('id');
```

Where the selectedId is undefined and for that reason, it thinks it has to select the folder again.

Reply:
yes because the jstree reloaded and there is no any selected folder, so in this case we don't need apply filter again as it's already there,A need only select the folder